packages/did/src/Did.utils.ts

```diff
@@ -135,29 +135,35 @@ export function validateUri(
     throw new TypeError(`DID string expected, got ${typeof input}`)
   }
   const { address, fragment } = parse(input as DidUri)
-  switch (expectType) {
-    // for backwards compatibility with previous implementations, `false` maps to `Did` while `true` maps to `undefined`.
-    // @ts-ignore
-    case false:
-    case 'Did':
-      if (fragment)
-        throw new SDKErrors.DidError(
-          'Expected a Kilt DidUri but got a DidResourceUri (containing a #fragment)'
-        )
-      break
-    case 'ResourceUri':
-      if (!fragment)
-        throw new SDKErrors.DidError(
-          'Expected a Kilt DidResourceUri (containing a #fragment) but got a DidUri'
-        )
-      break
-    default:
-      break
+
+  if (
+    fragment &&
+    (expectType === 'Did' ||
+      // for backwards compatibility with previous implementations, `false` maps to `Did` while `true` maps to `undefined`.
+      (typeof expectType === 'boolean' && expectType === false))
+  ) {
+    throw new SDKErrors.DidError(
+      'Expected a Kilt DidUri but got a DidResourceUri (containing a #fragment)'
+    )
+  }
+
+  if (!fragment && expectType === 'ResourceUri') {
+    throw new SDKErrors.DidError(
+      'Expected a Kilt DidResourceUri (containing a #fragment) but got a DidUri'
+    )
   }
 
   DataUtils.verifyKiltAddress(address)
 }
 
+/**
+ * Internal: derive the address part of the DID when it is created from authentication key.
```

Comment:
Can use [@internal](https://tsdoc.org/pages/tags/internal/) for internal functions. This will even remove it from d.ts files.

Reply:
ESLint: Invalid JSDoc tag name "internal".(jsdoc/check-tag-names)

Reply:
While this is a nice thing to have, I’d rather have it introduced and consistently applied to the codebase in a separate PR.